### DISCUSSION TITLE
How to do comparisons with tuple comparison?

### DISCUSSION THREAD

**Discussion by cs-clarence** [diesel-rs/diesel#3750] [Q&A]
How do you express this query in diesel query builder?

```sql
SELECT * FROM workspaces WHERE (created_at, id) < ($1, $2) ORDER BY created_at DESC;
```

I'm trying to implement a cursor based pagination with `UUID`s and `TIMESTAMP`s. The part where I'm having trouble is the WHERE clause which uses a tuple comparison and I don't know how to express that in diesel query builder. I haven't found anything in the docs.

I've tried doing this:

```rs
let query = workspaces::table
  .order_by(workspaces::created_at.desc())
  .filter(
    (workspaces::created_at, workspaces::id)
      .lt(after.created_at, after.id),
  )
  .into_boxed();
```

But I'm getting an error

```
the method `lt` exists for tuple `(created_at, id)`, but its trait bounds were not satisfied
the following trait bounds were not satisfied
```

**Comment by weiznich** [comment]
Duplicate of https://github.com/diesel-rs/diesel/discussions/3723
